### PR TITLE
[refer #382] Add SessionIdPeerFeature

### PR DESCRIPTION
**Pull request by knizhnik**

### CODE REVIEW

File: src/main/scala/scorex/core/network/peer/SessionIdPeerFeature.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+object SessionIdPeerFeatureSerializer extends ScorexSerializer[SessionIdPeerFeature] {
+  override def serialize(obj: SessionIdPeerFeature, w: Writer): Unit = {
+    w.putInt(obj.networkMagic.size)
+    w.putBytes(obj.networkMagic)
+    w.putLong(obj.sessionId)
+  }
+
+  override def parse(r: Reader): SessionIdPeerFeature = {
+    val networkMagic = r.getBytes(r.getInt())
```
Comment (kushti) [inline src/main/scala/scorex/core/network/peer/SessionIdPeerFeature.scala:35]: better to use putUInt / getUInt

File: src/main/scala/scorex/core/network/NetworkController.scala [hunk 2]
```diff
@@ -290,13 +291,11 @@ class NetworkController(settings: NetworkSettings,
           s"New outgoing connection to ${connectionId.remoteAddress} established (bound to local ${connectionId.localAddress})"
       }
     }
-
     val isLocal = connectionId.remoteAddress.getAddress.isSiteLocalAddress ||
       connectionId.remoteAddress.getAddress.isLoopbackAddress
-    val peerFeatures =
-      if (isLocal) scorexContext.features :+ LocalAddressPeerFeature(
+    var peerFeatures = scorexContext.features :+ mySessionIdFeature
```
Comment (kushti) [inline src/main/scala/scorex/core/network/NetworkController.scala:296]: peerFeatures can be immutable (val) still

File: src/main/scala/scorex/core/network/peer/SessionIdPeerFeature.scala [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    val sessionId = r.getLong()
+    SessionIdPeerFeature(networkMagic, sessionId)
+  }
+}
```
Comment (kushti) [inline src/main/scala/scorex/core/network/peer/SessionIdPeerFeature.scala:39]: Please do a test for serialization (round-trip at least)

**Review by kushti — CHANGES_REQUESTED** [review]

File: src/main/scala/scorex/core/network/peer/SessionIdPeerFeature.scala (hunk 1 above)
Comment (knizhnik) [reply src/main/scala/scorex/core/network/peer/SessionIdPeerFeature.scala:35]: getUInt() returns Long and can not be passed to getBytes

Comment (kushti) [reply src/main/scala/scorex/core/network/peer/SessionIdPeerFeature.scala:35]: Actually, network magic is always about 4 bytes, so no need to pass this value

**Review by kushti — APPROVED** [review]

**Pull request merged**